✨: Sign in with Apple

qlife1146 opened the pull request:
## 🔗 관련 이슈

close #2 

## ✅ 체크리스트
- [x] **base 브랜치를 develop으로 설정했나요?**
- [x] **작업 후 develop 브랜치를 Pull 받았나요?**
- [x] **PR의 라벨을 설정했나요?**
- [x] **assignee를 설정했나요?**
- [x] **reviewers를 설정했나요?**
- [x] **변경 사항에 대한 테스트를 진행했나요?**

## 🛠️ 작업 내용
> 변경한 핵심 내용을 간단히 요약해주세요.

- Add
  - Managers/AuthClient.swift - api와 주소를 가공
  - Presentation/Auth/LoginViewController.swift - 버튼 입력 감시, Reactor로 토큰 전달
  - Presentation/Auth/LoginReactor.swift - View의 액션 처리(인증, 상태, 라우팅 관리)
- Feat
  - Apple 로그인 구현

## 📸 스크린샷
> UI 관련 변경이 있다면 스크린샷을 첨부해주세요.
![Simulator Screen Recording - iPhone 16 Pro Max - 2025-08-22 at 11 45 54](https://github.com/user-attachments/assets/3eb75e31-341f-4d3c-9274-4601528c11f4)

## 💬 기타 참고사항
> 리뷰어가 알아야 할 내용이나 추가 설명이 있다면 여기에 적어주세요.

- 아직 DB에는 안 들어갑니다. 그래서 기가입자 판단을 못 합니다. 할 때마다 처음 온 것처럼

juseojo committed:
💡 Chore: 팀 설정

qlife1146 committed:
init

qlife1146 committed:
애플 로그인 사전 코드

qlife1146 committed:
2ba434d
📦: Managers/AuthClient.swift - 소셜 로그인에서 api키와 주소 관리
📦: Presentation/Auth/Login/LoginViewController - 소셜 로그인 버튼이 모여있는 화면(토큰을 Reactor로 전달)
📦: Presentation/Auth/Login/LoginReactor.swift - View의 액션을 처리(인증, 상태, 라우팅 관리)
✨: Apple 로그인 구현
📄: 주석 일부 추가

qlife1146 committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

qlife1146 committed:
🎨: 버튼에 심볼 삽입

qlife1146 committed:
✨: font 추가
🎨: font 적용
🔨: Lint 적용

Jaesung-Jung commented on Bragit/Presentation/Auth/Login/LoginViewController.swift:
```diff
@@ -0,0 +1,241 @@
+  // Sign in with Apple에서 권장(nonce 생성 및 해시 적용)
+
+  // nonce 생성(32자 랜덤 문자열 생성)
+  fileprivate func randomNonce(length: Int = 32) -> String {
+    precondition(length > 0)
+    let charSet: [Character] = Array("0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyz-._")
+    var result = ""
+    var remaining = length
+
+    while remaining > 0 {
+      var random: UInt8 = 0
+      let status = SecRandomCopyBytes(kSecRandomDefault, 1, &random)
+      if status != errSecSuccess { fatalError("Unable to generate nonce.") }
+      if random < charSet.count {
+        result.append(charSet[Int(random % UInt8(charSet.count))])
+        remaining -= 1
+      }
+    }
+    return result
+  }
+
+  // nonce -> 해시 변환
+  fileprivate func sha256(_ input: String) -> String {
+    let inputData = Data(input.utf8)
+    let hashed = SHA256.hash(data: inputData)
+    return hashed.compactMap { String(format: "%02x", $0) }.joined()
+  }
```
nonce 생성 로직은 Reactor로 들어가면 더 좋지 않을까요?

qlife1146 replied:
> nonce 생성 로직은 Reactor로 들어가면 더 좋지 않을까요?

한번 알아보겠습니다.

juseojo commented on Bragit/Presentation/Auth/Login/LoginReactor.swift:
```diff
@@ -0,0 +1,100 @@
+
+  // Supabase 의존성 (현재 코드와 동일한 생성값을 기본 주입)
+  private let supabase: SupabaseClient
+  init(supabase: SupabaseClient = AuthClient.shared) {
```
AuthClient를 swift-dependency로 관리하면 싱글톤을 사용하며 생기는 문제점을 해결하고
여러 좋은 점들이 생길거 같습니다.

swift-dependency 사용은
https://github.com/Brag-it/Bragit/pull/15 해당 PR 에서 참고하면 좋을거 같습니다.

Pull 받으시면 생기는 DependencyKeys 파일에
DependencyKey를 정의하고 liveValue로 SupabaseClient를 url과 api를 함께 생성하고
사용처(현재 이곳)에서
 @Dependency(\.디펜던시키이름) var supabase
 이런 식 으로 정의하면 편리하게 사용가능 합니다!
 
 쓰고나서 생각해보니 supabase dependency 코드는 제가 구현 해도 되는 영역같아서
디펜던시만 구현해서 올리겠습니디.

qlife1146 replied:
네 확인 중입니다.

qlife1146 committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Lee-Tae-Yun reviewed (approved):
카카오톡 로그인 버튼 색상은 카카오톡의 노랑색인가요 아니면 임의의 색을 넣으신 걸까요?? 
카카오톡의 색상이맞다면 색상을 에셋에 추가해서 하면 좋을거 같습니다!

hbrheg commented:
카카오톡 노란색 맞습니다!!
피그마에 추가해두겟습니당


2025년 8월 22일 (금) 오후 10:29, 이태윤 ***@***.***>님이 작성:

> ***@***.**** approved this pull request.
>
> 카카오톡 로그인 버튼 색상은 카카오톡의 노랑색인가요 아니면 임의의 색을 넣으신 걸까요??
> 카카오톡의 색상이맞다면 색상을 에셋에 추가해서 하면 좋을거 같습니다!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Brag-it/Bragit/pull/14#pullrequestreview-3144521981>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BWEVSUPAVLP6R5L7C7USUID3O4LMTAVCNFSM6AAAAACEQQ3E5WVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTCNBUGUZDCOJYGE>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

qlife1146 committed:
🔨: DependencyKeys에 Lint 경고 수정
✨: swift-dependency 도입

qlife1146 committed:
🔨: Lint 경고 수정

qlife1146 committed:
🚀: nonce 로직을 reactor로 이동

qlife1146 committed:
🔨: 주석 삭제

juseojo commented on Bragit/Model/DependencyKeys.swift:
```diff
@@ -5,9 +5,8 @@
-import Foundation
-
 import Dependencies
+import Foundation
```
https://github.com/StyleShare/swift-style-guide

저희 가이드에 따르면 내장 프레임 워크가 상단으로 가야하고 한칸 구분되야 해서 이전처럼 수정 가능할까요?

qlife1146 replied:
수정했습니다. 추가적으로 제가 쓴 다른 것들도 맞춰서 수정했습니다.

juseojo commented on Bragit/Managers/Auth/AuthClient.swift:
```diff
@@ -0,0 +1,72 @@
+  }
+}
+
+extension AuthClient: DependencyKey {
```
DependencyKey와 values를 Model의 DependencyKeys에 코드를 모아놓으면 좋을거 같아요!

qlife1146 replied:
수정한대로 옮기면 될까요?

juseojo commented on Bragit/Managers/Auth/AuthClient.swift:
```diff
@@ -0,0 +1,72 @@
+  var signInWithApple: (_ idToken: String, _ nonce: String) async throws -> Void
+}
+
+extension AuthClient {
```
코드가 이전보다 깔끔하고 명확해져서 좋은거 같습니당

qlife1146 committed:
🔨: 라이브러리 정렬
🔨: 주석 삭제
🔨: Lint 경고 수정
🔨: Dependency Key, values를 Model/DependencyKeys로 이동
